# Comprehensive List of Detected Project Files

This file lists all data files detected as inputs to or outputs from the Jupyter notebooks in this repository.

## Detected Files
- [`data/huh7/SraRunTable.csv`](data/huh7/SraRunTable.csv): Comma-separated values file with tabular data.
- [`data/processed_files/apsp_nonhs.pkl.gz`](data/processed_files/apsp_nonhs.pkl.gz): Processed file generated via Bash or Python command in notebook.
- [`data/processed_files/asp_nonhs.pkl.gz`](data/processed_files/asp_nonhs.pkl.gz): Processed file generated via Bash or Python command in notebook.
- [`data/processed_files/combined.bed`](data/processed_files/combined.bed): Genomic features in BED format.
- [`data/processed_files/consensus.fa`](data/processed_files/consensus.fa): Consensus HBV pgRNA sequences.
- [`data/processed_files/consensus.fas`](data/processed_files/consensus.fas): Consensus HBV pgRNA sequences.
- [`data/processed_files/consensus_flank255_510.fa`](data/processed_files/consensus_flank255_510.fa): Consensus HBV pgRNA sequences.
- [`data/processed_files/consensus_hg19.fa`](data/processed_files/consensus_hg19.fa): Consensus HBV pgRNA sequences.
- [`data/processed_files/consensus_splice_sites.bed`](data/processed_files/consensus_splice_sites.bed): Consensus splice site coordinates.
- [`data/processed_files/consensus_splice_sites.bed.gz`](data/processed_files/consensus_splice_sites.bed.gz): Consensus splice site coordinates.
- [`data/processed_files/conss.all.stats`](data/processed_files/conss.all.stats): Splice site statistics across clusters.
- [`data/processed_files/conss.bed`](data/processed_files/conss.bed): Genomic features in BED format.
- [`data/processed_files/conss.cds.stats`](data/processed_files/conss.cds.stats): Splice site statistics across clusters.
- [`data/processed_files/conss_nonhs.all.stats`](data/processed_files/conss_nonhs.all.stats): Splice site statistics across clusters.
- [`data/processed_files/conss_nonhs.bed`](data/processed_files/conss_nonhs.bed): Genomic features in BED format.
- [`data/processed_files/conss_nonhs.cds.stats`](data/processed_files/conss_nonhs.cds.stats): Splice site statistics across clusters.
- [`data/processed_files/cosi.hbv.txt`](data/processed_files/cosi.hbv.txt): Tab-delimited text file with processed results.
- [`data/processed_files/cosi.pkl.gz`](data/processed_files/cosi.pkl.gz): Wide-format coSI scores for HBV splice donor and acceptor pairs.
- [`data/processed_files/cosi_long.pkl.gz`](data/processed_files/cosi_long.pkl.gz): Long-format coSI scores for HBV splice donor and acceptor sites.
- [`data/processed_files/crosstab_acceptors_perc.csv`](data/processed_files/crosstab_acceptors_perc.csv): Cluster-level percentage of true splice site labels (donor/acceptor).
- [`data/processed_files/crosstab_acceptors_perc.pkl.gz`](data/processed_files/crosstab_acceptors_perc.pkl.gz): Cluster-level percentage of true splice site labels (donor/acceptor).
- [`data/processed_files/crosstab_acceptors_perc_nonhs.csv`](data/processed_files/crosstab_acceptors_perc_nonhs.csv): Cluster-level percentage of true splice site labels (donor/acceptor).
- [`data/processed_files/crosstab_acceptors_perc_nonhs.pkl.gz`](data/processed_files/crosstab_acceptors_perc_nonhs.pkl.gz): Cluster-level percentage of true splice site labels (donor/acceptor).
- [`data/processed_files/crosstab_donors_perc.csv`](data/processed_files/crosstab_donors_perc.csv): Cluster-level percentage of true splice site labels (donor/acceptor).
- [`data/processed_files/crosstab_donors_perc.pkl.gz`](data/processed_files/crosstab_donors_perc.pkl.gz): Cluster-level percentage of true splice site labels (donor/acceptor).
- [`data/processed_files/crosstab_donors_perc_nonhs.csv`](data/processed_files/crosstab_donors_perc_nonhs.csv): Cluster-level percentage of true splice site labels (donor/acceptor).
- [`data/processed_files/crosstab_donors_perc_nonhs.pkl.gz`](data/processed_files/crosstab_donors_perc_nonhs.pkl.gz): Cluster-level percentage of true splice site labels (donor/acceptor).
- [`data/processed_files/cstats.pkl.gz`](data/processed_files/cstats.pkl.gz): Splice site statistics across clusters.
- [`data/processed_files/cstats_nonhs.pkl.gz`](data/processed_files/cstats_nonhs.pkl.gz): Splice site statistics across clusters.
- [`data/processed_files/dpsp_nonhs.pkl.gz`](data/processed_files/dpsp_nonhs.pkl.gz): Processed file generated via Bash or Python command in notebook.
- [`data/processed_files/dsp_nonhs.pkl.gz`](data/processed_files/dsp_nonhs.pkl.gz): Processed file generated via Bash or Python command in notebook.
- [`data/processed_files/exonic_acceptor.ids`](data/processed_files/exonic_acceptor.ids): Splice donor/acceptor site coordinates.
- [`data/processed_files/exonic_donor.ids`](data/processed_files/exonic_donor.ids): Splice donor/acceptor site coordinates.
- [`data/processed_files/exonicss.hbv.txt`](data/processed_files/exonicss.hbv.txt): Mapped coSI scores to exonic splice sites.
- [`data/processed_files/exonicss.ids`](data/processed_files/exonicss.ids): Mapped coSI scores to exonic splice sites.
- [`data/processed_files/exonicss.txt`](data/processed_files/exonicss.txt): Mapped coSI scores to exonic splice sites.
- [`data/processed_files/exonicss_genes.fa`](data/processed_files/exonicss_genes.fa): Mapped coSI scores to exonic splice sites.
- [`data/processed_files/exonicss_genes.split.30mer.fa`](data/processed_files/exonicss_genes.split.30mer.fa): Mapped coSI scores to exonic splice sites.
- [`data/processed_files/exonicss_mc3.bed`](data/processed_files/exonicss_mc3.bed): Mapped coSI scores to exonic splice sites.
- [`data/processed_files/exonicss_mc3.fa`](data/processed_files/exonicss_mc3.fa): Mapped coSI scores to exonic splice sites.
- [`data/processed_files/exonicss_mc5.bed`](data/processed_files/exonicss_mc5.bed): Mapped coSI scores to exonic splice sites.
- [`data/processed_files/exonicss_mc5.fa`](data/processed_files/exonicss_mc5.fa): Mapped coSI scores to exonic splice sites.
- [`data/processed_files/exonicss_nonhs.hbv.txt`](data/processed_files/exonicss_nonhs.hbv.txt): Mapped coSI scores to exonic splice sites.
- [`data/processed_files/exonicss_nonhs.txt`](data/processed_files/exonicss_nonhs.txt): Mapped coSI scores to exonic splice sites.
- [`data/processed_files/exons.hs.bed`](data/processed_files/exons.hs.bed): Genomic features in BED format.
- [`data/processed_files/hbv.ss.bed`](data/processed_files/hbv.ss.bed): Genomic features in BED format.
- [`data/processed_files/hbvdb_pgrna.fa`](data/processed_files/hbvdb_pgrna.fa): FASTA file containing nucleotide sequences.
- [`data/processed_files/hbvdb_pgrna.split.30mer.fa`](data/processed_files/hbvdb_pgrna.split.30mer.fa): FASTA file containing nucleotide sequences.
- [`data/processed_files/hbvdb_pgrna_mc3.bed`](data/processed_files/hbvdb_pgrna_mc3.bed): Genomic features in BED format.
- [`data/processed_files/hbvdb_pgrna_mc3.pkl.gz`](data/processed_files/hbvdb_pgrna_mc3.pkl.gz): Processed file generated via Bash or Python command in notebook.
- [`data/processed_files/hbvdb_pgrna_mc3_high.fa`](data/processed_files/hbvdb_pgrna_mc3_high.fa): FASTA file containing nucleotide sequences.
- [`data/processed_files/hbvdb_pgrna_mc3_low.fa`](data/processed_files/hbvdb_pgrna_mc3_low.fa): FASTA file containing nucleotide sequences.
- [`data/processed_files/hbvdb_pgrna_mc5.bed`](data/processed_files/hbvdb_pgrna_mc5.bed): Genomic features in BED format.
- [`data/processed_files/hbvdb_pgrna_mc5.pkl.gz`](data/processed_files/hbvdb_pgrna_mc5.pkl.gz): Processed file generated via Bash or Python command in notebook.
- [`data/processed_files/hbvdb_pgrna_mc5_high.fa`](data/processed_files/hbvdb_pgrna_mc5_high.fa): FASTA file containing nucleotide sequences.
- [`data/processed_files/hbvdb_pgrna_mc5_low.fa`](data/processed_files/hbvdb_pgrna_mc5_low.fa): FASTA file containing nucleotide sequences.
- [`data/processed_files/hg19_genes.split.30mer.fa`](data/processed_files/hg19_genes.split.30mer.fa): FASTA file containing nucleotide sequences.
- [`data/processed_files/hg19_mc3.bed`](data/processed_files/hg19_mc3.bed): Genomic features in BED format.
- [`data/processed_files/hg19_mc3.bed2`](data/processed_files/hg19_mc3.bed2): Processed file generated via Bash or Python command in notebook.
- [`data/processed_files/hg19_mc3.fa`](data/processed_files/hg19_mc3.fa): FASTA file containing nucleotide sequences.
- [`data/processed_files/hg19_mc5.bed`](data/processed_files/hg19_mc5.bed): Genomic features in BED format.
- [`data/processed_files/hg19_mc5.bed2`](data/processed_files/hg19_mc5.bed2): Processed file generated via Bash or Python command in notebook.
- [`data/processed_files/hg19_mc5.fa`](data/processed_files/hg19_mc5.fa): FASTA file containing nucleotide sequences.
- [`data/processed_files/intron.zeta.bed`](data/processed_files/intron.zeta.bed): coSI scores for splice sites.
- [`data/processed_files/lib.txt`](data/processed_files/lib.txt): Tab-delimited text file with processed results.
- [`data/processed_files/map.txt`](data/processed_files/map.txt): Tab-delimited text file with processed results.
- [`data/processed_files/mc3.bed`](data/processed_files/mc3.bed): Genomic features in BED format.
- [`data/processed_files/mc5.bed`](data/processed_files/mc5.bed): Genomic features in BED format.
- [`data/processed_files/mgen-7-492-s002.xlsx`](data/processed_files/mgen-7-492-s002.xlsx): Supplementary file from our previous study
- [`data/processed_files/nonss.bed`](data/processed_files/nonss.bed): Genomic features in BED format.
- [`data/processed_files/sj.out.tab`](data/processed_files/sj.out.tab): STAR splice junction output.
- [`data/processed_files/splice_acceptor_sj.bed`](data/processed_files/splice_acceptor_sj.bed): Splice donor/acceptor site coordinates.
- [`data/processed_files/splice_acceptors.bed`](data/processed_files/splice_acceptors.bed): Splice donor/acceptor site coordinates.
- [`data/processed_files/splice_acceptors_.bed`](data/processed_files/splice_acceptors_.bed): Splice donor/acceptor site coordinates.
- [`data/processed_files/splice_acceptors_inferred.bed`](data/processed_files/splice_acceptors_inferred.bed): Splice donor/acceptor site coordinates.
- [`data/processed_files/splice_acceptors_inferred.bed2`](data/processed_files/splice_acceptors_inferred.bed2): Splice donor/acceptor site coordinates.
- [`data/processed_files/splice_donor_sj.bed`](data/processed_files/splice_donor_sj.bed): Splice donor/acceptor site coordinates.
- [`data/processed_files/splice_donors.bed`](data/processed_files/splice_donors.bed): Splice donor/acceptor site coordinates.
- [`data/processed_files/splice_donors_.bed`](data/processed_files/splice_donors_.bed): Splice donor/acceptor site coordinates.
- [`data/processed_files/splice_donors_inferred.bed`](data/processed_files/splice_donors_inferred.bed): Splice donor/acceptor site coordinates.
- [`data/processed_files/splice_donors_inferred.bed2`](data/processed_files/splice_donors_inferred.bed2): Splice donor/acceptor site coordinates.
- [`data/processed_files/ss.bed`](data/processed_files/ss.bed): Genomic features in BED format.
- [`data/processed_files/ss.hbv.bed`](data/processed_files/ss.hbv.bed): Genomic features in BED format.
- [`data/processed_files/ss.hbv.bed2`](data/processed_files/ss.hbv.bed2): Processed file generated via Bash or Python command in notebook.
- [`data/processed_files/ss.hs.bed`](data/processed_files/ss.hs.bed): Genomic features in BED format.
- [`data/processed_files/stringtie.bed`](data/processed_files/stringtie.bed): Genomic features in BED format.
- [`data/processed_files/stringtie.gtf`](data/processed_files/stringtie.gtf): Transcript annotation file in GTF format.
- [`data/processed_files/tcons.txt`](data/processed_files/tcons.txt): Tab-delimited text file with processed results.
- [`data/processed_files/track.lol.txt`](data/processed_files/track.lol.txt): Tab-delimited text file with processed results.
- [`ref/hbvdb/pgrna/pgrna_flank200.txt`](ref/hbvdb/pgrna/pgrna_flank200.txt): Tab-delimited text file with processed results.
- [`ref/hbvdb/pgrna/pgrna_flank200_AG-AA.txt`](ref/hbvdb/pgrna/pgrna_flank200_AG-AA.txt): Tab-delimited text file with processed results.
- [`ref/hbvdb/pgrna/pgrna_flank200_GT-AA.txt`](ref/hbvdb/pgrna/pgrna_flank200_GT-AA.txt): Tab-delimited text file with processed results.
- [`ref/hg19_genes.fa`](ref/hg19_genes.fa): FASTA file containing nucleotide sequences.
- [`ref/ipsa_index/hbv.gfx`](ref/ipsa_index/hbv.gfx): Transcript annotation file generated by IPSA pipeline (GFX format).
- [`ref/ipsa_index/hbv.gtf`](ref/ipsa_index/hbv.gtf): Transcript annotation file in GTF format.
- [`ref/pgrna/B2/B2.pgrna.fa.map`](ref/pgrna/B2/B2.pgrna.fa.map): Coordinate mapping file in MAFFT mapout format.
- [`ref/pgrna/C2/C2.pgrna.fa.map`](ref/pgrna/C2/C2.pgrna.fa.map): Coordinate mapping file in MAFFT mapout format.
- [`ref/pgrna/D3/D3.pgrna.fa.map`](ref/pgrna/D3/D3.pgrna.fa.map): Coordinate mapping file in MAFFT mapout format.
- [`ref/utr.hs.bed`](ref/utr.hs.bed): Genomic features in BED format.
- [`results/data/Percent_HBV_RNA.csv`](results/data/Percent_HBV_RNA.csv): Final proportions of spliced HBV RNAs used in results.
- [`results/data/coSI.csv`](results/data/coSI.csv): Final coSI scores for HBV splice sites used in results.
- [`results/figures/fig1/cosi_splice_variants.pdf`](results/figures/fig1/cosi_splice_variants.pdf): Generated plot or figure output.
- [`results/figures/fig1/percent_splice_variants.pdf`](results/figures/fig1/percent_splice_variants.pdf): Generated plot or figure output.
- [`results/figures/fig2/umap.acceptor.leiden.png`](results/figures/fig2/umap.acceptor.leiden.png): Generated plot or figure output.
- [`results/figures/fig2/umap.acceptor.png`](results/figures/fig2/umap.acceptor.png): Generated plot or figure output.
- [`results/figures/fig2/umap.donor.png`](results/figures/fig2/umap.donor.png): Generated plot or figure output.
- [`results/figures/fig2/umap.donors.leiden.png`](results/figures/fig2/umap.donors.leiden.png): Generated plot or figure output.
- [`results/figures/fig3/cosi_acceptor.leiden.png`](results/figures/fig3/cosi_acceptor.leiden.png): Generated plot or figure output.
- [`results/figures/fig3/cosi_donor.leiden.png`](results/figures/fig3/cosi_donor.leiden.png): Generated plot or figure output.
- [`results/figures/fig3/typical_splice_sites.png`](results/figures/fig3/typical_splice_sites.png): Generated plot or figure output.
- [`results/figures/fig4/umap.logit_acceptors.leiden.png`](results/figures/fig4/umap.logit_acceptors.leiden.png): Generated plot or figure output.
- [`results/figures/fig4/umap.logit_acceptors.png`](results/figures/fig4/umap.logit_acceptors.png): Generated plot or figure output.
- [`results/figures/fig5/conservation_acceptors.png`](results/figures/fig5/conservation_acceptors.png): Generated plot or figure output.
- [`results/figures/fig5/conservation_donors.png`](results/figures/fig5/conservation_donors.png): Generated plot or figure output.
- [`results/figures/fig5/logit_acceptors.png`](results/figures/fig5/logit_acceptors.png): Generated plot or figure output.
- [`results/figures/fig5/logit_donors.png`](results/figures/fig5/logit_donors.png): Generated plot or figure output.
- [`results/figures/fig5/openspliceai_acceptors.png`](results/figures/fig5/openspliceai_acceptors.png): Generated plot or figure output.
- [`results/figures/fig5/openspliceai_donors.png`](results/figures/fig5/openspliceai_donors.png): Generated plot or figure output.
- [`results/figures/fig6/usage_comparison_mc3.pdf`](results/figures/fig6/usage_comparison_mc3.pdf): Generated plot or figure output.
- [`results/figures/fig6/usage_comparison_mc5.pdf`](results/figures/fig6/usage_comparison_mc5.pdf): Generated plot or figure output.
- [`results/figures/figs1/cosi_splice_variants-sm.pdf`](results/figures/figs1/cosi_splice_variants-sm.pdf): Generated plot or figure output.
- [`results/figures/figs1/percent_rna-sm.pdf`](results/figures/figs1/percent_rna-sm.pdf): Generated plot or figure output.
- [`results/figures/figs1/percent_splice_variants_HBV_genotype.pdf`](results/figures/figs1/percent_splice_variants_HBV_genotype.pdf): Generated plot or figure output.
- [`results/figures/figs1/sp6-sm.pdf`](results/figures/figs1/sp6-sm.pdf): Generated plot or figure output.
- [`results/figures/figs2/umap.acceptor.leiden.png`](results/figures/figs2/umap.acceptor.leiden.png): Generated plot or figure output.
- [`results/figures/figs2/umap.acceptor.png`](results/figures/figs2/umap.acceptor.png): Generated plot or figure output.
- [`results/figures/figs2/umap.donor.png`](results/figures/figs2/umap.donor.png): Generated plot or figure output.
- [`results/figures/figs2/umap.donors.leiden.png`](results/figures/figs2/umap.donors.leiden.png): Generated plot or figure output.
- [`results/figures/figs3/cosi_acceptor.leiden.png`](results/figures/figs3/cosi_acceptor.leiden.png): Generated plot or figure output.
- [`results/figures/figs3/cosi_donor.leiden.png`](results/figures/figs3/cosi_donor.leiden.png): Generated plot or figure output.
- [`results/figures/figs3/typical_splice_sites.png`](results/figures/figs3/typical_splice_sites.png): Generated plot or figure output.
